user_model.ml

```diff
@@ -94,32 +106,74 @@ let cookie_of_json = function
               expires_in;
               uuid = None;
               created_at = Option.get created_at;
+              last_access = created_at;
+              user_agent = None;
             }
       | _ -> Error (`Msg "invalid json for cookie"))
   | _ -> Error (`Msg "invalid json for cookie")
 
-let cookie_to_string cookie = Yojson.Basic.to_string (cookie_to_json cookie)
-
-let cookie_of_string (s : string) : cookie option =
-  try
-    let json = Yojson.Basic.from_string s in
-    let open Yojson.Basic.Util in
-    let name = json |> member "name" |> to_string in
-    let value = json |> member "value" |> to_string in
-    let expires_in = json |> member "expires_in" |> to_int in
-    let uuid =
-      match json |> member "uuid" with `String uuid -> Some uuid | _ -> None
-    in
-    let created_at =
+let cookie_of_json = function
+  | `Assoc xs -> (
       match
-        json |> member "created_at" |> to_string
-        |> Utils.TimeHelper.ptime_of_string
+        ( get "name" xs,
+          get "value" xs,
+          get "expires_in" xs,
+          get "uuid" xs,
+          get "created_at" xs,
+          get "last_access" xs,
+          get "user_agent" xs )
       with
-      | Ok ptime -> Some ptime
-      | Error _ -> None
-    in
-    Some { name; value; expires_in; uuid; created_at = Option.get created_at }
-  with _ -> None
+      | ( Some (`String name),
+          Some (`String value),
+          Some (`Int expires_in),
+          Some (`String uuid),
+          Some (`String created_at_str),
+          Some (`String last_access_str),
+          Some (`String user_agent) ) ->
```

Comment:
error handling for created_at and last_access --- the `Option.get` raises if the value is `None`.

Reply:
done in 3ae62de